### PR TITLE
Add joboffer model

### PR DESCRIPTION
**Pull request by martinpfister**
this PR adds the joboffer model.

### CODE REVIEW

**Comment by janikvonrotz** [comment]
@martinpfister Thanks for the PR, I will review it on monday :)

File: job_portal_base/static/description/icon.png
Comment (janikvonrotz) [inline job_portal_base/static/description/icon.png:1]: @martinpfister I have used this Odoo Icon Builder: https://spilymp.github.io/ibo/

We can use your icon, but can you scale it to 100 x 100 ?

**Comment by janikvonrotz** [comment]
@martinpfister Can you split this PR into two modules:

**job_portal_base** (depends on base  and already exists)

This module contains the joboffer model and all view required to show the app.

**job_portal_sale** (depends on job_portal_base and sale)

This module extends the joboffer model and views with actions and buttons to create a sale order.

File: job_portal_base/static/description/icon.png
Comment (martinpfister) [reply job_portal_base/static/description/icon.png:1]: i replaced the icon with one i generated with the tool you provided. not perfect, can replace later again.

File: job_portal_base/views/joboffer.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_jobportal_joboffer_form" model="ir.ui.view">
+        <field name="name">soz.joboffer.form</field>
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/views/joboffer.xml]: done for all joboffer views

File: job_portal_base/views/joboffer.xml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        <field name="model">soz.joboffer</field>
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/views/joboffer.xml]: yes. fixed.

File: job_portal_base/__manifest__.py [hunk 3]
```diff
@@ -5,13 +5,22 @@
     """,
     "author": "sozialinfo.ch",
     "website": "https://www.sozialinfo.ch",
-    "category": "Services",
+    "category": "Sozialinfo",
     "version": "16.0.1.0.0",
     "license": "AGPL-3",
-    "depends": ["base", "product"],
-    "data": ["views/product_template.xml"],
+    "depends": ["base", "mail", "sale", "job_portal_category"],
```
Comment (janikvonrotz) [inline job_portal_base/__manifest__.py:11]: The job_portal_base, must depend only on the base modul.

File: job_portal_base/data/sequence.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<odoo>
+    <data noupdate="1">
+        <record id="job_portal_base.joboffer_ref_sequence" model="ir.sequence">
+            <field name="name">ST-Nummer</field>
```
Comment (janikvonrotz) [inline job_portal_base/data/sequence.xml:4]: What does ST stand for? What is the english definition here?

File: job_portal_base/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+job_portal_admin,Job Portal Admin,model_job_portal_joboffer,group_job_portal_admin,1,1,1,1
```
Comment (janikvonrotz) [inline job_portal_base/security/ir.model.access.csv:2]: Use manager instead of admin.

File: job_portal_base/security/job_portal_groups.xml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<odoo>
+    <record id="group_job_portal_admin" model="res.groups">
+        <field name="name">Stellenportal Admin</field>
```
Comment (janikvonrotz) [inline job_portal_base/security/job_portal_groups.xml:3]: Job Portal Administrator

File: job_portal_base/security/job_portal_groups.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    </record>
+    <record id="group_job_portal_user" model="res.groups">
+        <field name="name">Stellenportal User</field>
```
Comment (janikvonrotz) [inline job_portal_base/security/job_portal_groups.xml:6]: Job Portal User

File: job_portal_base/static/description/icon.png
Comment (janikvonrotz) [inline job_portal_base/static/description/icon.png:1]: Can you use the sozialinfo.ch icon as proposed by David in #2 ?

File: job_portal_base/views/joboffer_view_act_window.xml
Comment (janikvonrotz) [inline job_portal_base/views/joboffer_view_act_window.xml:1]: Can you create a job_offer.xml and move everything into this file?

By everything I mean: joboffer_view_act_window.xml, joboffer_view_from.xml, joboffer_view_list.xml, joboffer_view_search.xml, menu.xml

One file for each model.

File: job_portal_base/views/joboffer_view_form.xml [hunk 8]
```diff
@@ -0,0 +1,177 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="job_portal_base.joboffer_form" model="ir.ui.view">
+            <field name="name">job_portal.joboffer.form</field>
+            <field name="model">job_portal.joboffer</field>
+            <field name="arch" type="xml">
+                <form>
+                    <header>
+                        <!--
```
Comment (janikvonrotz) [inline job_portal_base/views/joboffer_view_form.xml:10]: Comments must be removed before merge.

File: job_portal_category/.gitignore
Comment (janikvonrotz) [inline job_portal_category/.gitignore:1]: The .gitignore file already exists in the root.

File: job_portal_category/__init__.py [hunk 9]
```diff
@@ -0,0 +1,2 @@
+from . import controllers
```
Comment (janikvonrotz) [inline job_portal_category/__init__.py:1]: A controller is not necessary.

File: job_portal_category/security/ir.model.access.csv [hunk 10]
```diff
@@ -0,0 +1,9 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+job_portal_admin,Job Portal Admin,model_job_portal_working_area,group_job_portal_admin,1,1,1,1
```
Comment (janikvonrotz) [inline job_portal_category/security/ir.model.access.csv:2]: Use manager here as well.

File: job_portal_base/models/joboffer.py [hunk 11]
```diff
@@ -0,0 +1,174 @@
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class JobOffer(models.Model):
+    _name = "job_portal.joboffer"
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = 'Stellenangebot'
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/models/joboffer.py]: fixed

File: job_portal_base/__manifest__.py [hunk 12]
```diff
@@ -5,13 +5,22 @@
...
+        'data/sequence.xml',
+        'views/joboffer_view_act_window.xml',
+        'views/menu.xml',
+        'views/joboffer_view_search.xml',
+        'views/joboffer_view_list.xml',
+        'views/joboffer_view_form.xml',
+    ],
     "installable": True,
-    "application": False,
+    "application": True,
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/__manifest__.py:28]: learned like this in my course :p why was it set to False?

File: job_portal_base/data/sequence.xml (hunk 4 above)
Comment (martinpfister) [reply job_portal_base/data/sequence.xml]: ST heisst Stellenangebot. Es ist unser Preffix für jeden Record.
Field Label ist korrigiert.

File: job_portal_base/views/joboffer_view_act_window.xml
Comment (martinpfister) [reply job_portal_base/views/joboffer_view_act_window.xml:1]: we talked about this? i would it very much prefer it that way.

File: job_portal_category/.gitignore
Comment (martinpfister) [reply job_portal_category/.gitignore:1]: fixed

File: job_portal_category/__init__.py (hunk 9 above)
Comment (martinpfister) [reply job_portal_category/__init__.py]: fixed

File: job_portal_base/views/joboffer_view_form.xml (hunk 8 above)
Comment (martinpfister) [reply job_portal_base/views/joboffer_view_form.xml]: fixed

File: job_portal_category/security/ir.model.access.csv (hunk 10 above)
Comment (martinpfister) [reply job_portal_category/security/ir.model.access.csv]: i have admin (can do everything), editor (can do everything except delete joboffer records), and user.
which one is the manager? editor i guess?

File: job_portal_base/static/description/icon.png
Comment (martinpfister) [reply job_portal_base/static/description/icon.png:1]: i created a subtask for this: https://sozialinfo.atlassian.net/browse/VS-143

File: job_portal_base/__manifest__.py (hunk 3 above)
Comment (martinpfister) [reply job_portal_base/__manifest__.py:11]: i created a separate subtask for this: https://sozialinfo.atlassian.net/browse/VS-142

Comment (martinpfister) [reply job_portal_base/__manifest__.py:11]: well, "mail" is used for the "chatter" (activity and notes). 
"sale" is used by the product_template model you added.

i can split the category attributes out of the model tough.

Comment (martinpfister) [reply job_portal_base/__manifest__.py]: i could remove sale and job_portal_category.

File: job_portal_base/static/description/icon.png
Comment (martinpfister) [reply job_portal_base/static/description/icon.png:1]: i used the same one as david.

File: job_portal_base/__manifest__.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (janikvonrotz) [reply job_portal_base/__manifest__.py:28]: @martinpfister Only set this to true for the job_portal_base module. This is the main app and the other modules are bridge modules or extension.

File: job_portal_base/__manifest__.py (hunk 3 above)
Comment (janikvonrotz) [reply job_portal_base/__manifest__.py]: @martinpfister You can use `product` instead of `sale`.

File: job_portal_base/views/joboffer_view_act_window.xml
Comment (janikvonrotz) [reply job_portal_base/views/joboffer_view_act_window.xml:1]: @martinpfister Yes I remember the conversion, then we had one module for multiple models.

I still highly  recommend to have one file for each model.

That seems to be the standard across all Odoo modules.

File: job_portal_category/security/ir.model.access.csv (hunk 10 above)
Comment (janikvonrotz) [reply job_portal_category/security/ir.model.access.csv]: @martinpfister The admin is usually called manager. My idea is to replace `admin` with `manager`.

File: job_portal_base/__manifest__.py (hunk 3 above)
Comment (martinpfister) [reply job_portal_base/__manifest__.py]: fixed

File: job_portal_base/__manifest__.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/__manifest__.py:28]: got it. fixed.

File: job_portal_base/views/joboffer_view_act_window.xml
Comment (martinpfister) [reply job_portal_base/views/joboffer_view_act_window.xml:1]: [i see](https://www.odoo.com/documentation/16.0/de/contributing/development/coding_guidelines.html)
it says splitting up the files for each model. it does not say you can split them up further ;)
i could create a sub-folder for each model? or.. can we do it later? i can create a task for it.

File: job_portal_base/data/sequence.xml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+<odoo>
+    <data noupdate="1">
+        <record id="job_portal_base.joboffer_ref_sequence" model="ir.sequence">
+            <field name="name">ST-Number</field>
+            <field name="code">job_portal.joboffer</field>
+            <field name="prefix">ST--</field>
```
Comment (martinpfister) [inline job_portal_base/data/sequence.xml:6]: ST-

Comment (martinpfister) [reply job_portal_base/data/sequence.xml:6]: fixed

File: job_portal_base/security/job_portal_base_groups.xml
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/security/job_portal_base_groups.xml:1]: Neuer Task erstellt: https://sozialinfo.atlassian.net/browse/VS-159

File: job_portal_category/security/ir.model.access.csv [hunk 14]
```diff
@@ -0,0 +1,9 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+job_portal_working_area_manager,Job Portal Category Manager,model_job_portal_working_area,group_job_portal_category_manager,1,1,1,1
+job_portal_working_area_user,Job Portal User,model_job_portal_working_area,group_job_portal_category_user,1,0,0,0
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_category/security/ir.model.access.csv:3]: Dafür wurde ein neuer Task erstellt: https://sozialinfo.atlassian.net/browse/VS-159

File: job_portal_category/security/job_portal_category_groups.xml
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_category/security/job_portal_category_groups.xml:1]: https://sozialinfo.atlassian.net/browse/VS-159

File: job_portal_base/views/menu.xml [hunk 15]
```diff
@@ -0,0 +1,13 @@
+<odoo>
+    <data>
+        <menuitem id="menu_job_portal_base"
+                  name="Stellenportal"
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/views/menu.xml:4]: fixed

File: job_portal_base/security/job_portal_base_groups.xml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<odoo>
+    <record id="group_job_portal_joboffer_manager" model="res.groups">
+        <field name="name">Job Portal: JobOffer Manager</field>
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/security/job_portal_base_groups.xml:3]: fixed

File: job_portal_base/models/joboffer.py [hunk 17]
```diff
@@ -0,0 +1,125 @@
+from odoo import api, fields, models, _
+
+
+class JobOffer(models.Model):
+    _name = "job_portal.joboffer"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Job Offer"
+
+    ref = fields.Char(string="ST-Number", required=True, default=lambda self: _("ST-#"))
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_base/models/joboffer.py:9]: fixed

File: job_portal_category/README.rst [hunk 18]
```diff
@@ -0,0 +1,26 @@
+.. image:: https://img.shields.io/badge/licence-GPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/gpl-3.0-standalone.html
+    :alt: License: GPL-3
+
+===============
+Job Portal Base
+===============
+
+Base for job portal modules.
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_category/README.rst:9]: fixed

File: job_portal_category/__manifest__.py [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+    "version": "16.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["base", "job_portal_base"],
+    "data": [
+        "security/job_portal_category_groups.xml",
+        "security/ir.model.access.csv",
+        "demo/demo_employments.xml",
+        "demo/demo_positions.xml",
+        "demo/demo_qualifications.xml",
+        "demo/demo_working_areas.xml",
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_category/__manifest__.py:18]: fixed

File: job_portal_sale/views/inherit_joboffer_view_search.xml [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+<odoo>
+    <data>
+        <record model="ir.ui.view" id="job_portal_sale.inherit_job_portal_base_joboffer_search">
+            <field name="name">Job Offers</field>
+            <field name="model">job_portal.joboffer</field>
+            <field name="inherit_id" ref="job_portal_base.joboffer_search"/>
+            <field name="arch" type="xml">
+                <field name="body" position="after">
+                    <separator/>
+                    <filter name="filter_is_created_by_gsi" string="von der GSI erstellt"
```
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_sale/views/inherit_joboffer_view_search.xml:11]: fixed

File: job_portal_sale/security/ir.model.access.csv
[the remark this answers is not in the document]
Comment (martinpfister) [reply job_portal_sale/security/ir.model.access.csv:1]: fixed
* on new branch feature/VS-162

**Review by janikvonrotz — APPROVED** [review]

**Comment by janikvonrotz** [comment]
@martinpfister Ich habe gemerged.

Ich wünsche dir schöne Ferien :+1:

**Pull request merged**